crates/ruff_linter/src/rules/flake8_type_checking/mod.rs

```diff
@@ -62,6 +63,20 @@ mod tests {
         Ok(())
     }
 
+    #[test_case(Rule::RuntimeCastValue, Path::new("TC006.py"))]
+    fn preview_rules(rule_code: Rule, path: &Path) -> Result<()> {
+        let snapshot = format!("preview__{}_{}", rule_code.as_ref(), path.to_string_lossy());
+        let diagnostics = test_path(
+            Path::new("flake8_type_checking").join(path).as_path(),
+            &settings::LinterSettings {
+                preview: PreviewMode::Enabled,
+                ..settings::LinterSettings::for_rule(rule_code)
+            },
+        )?;
+        assert_messages!(snapshot, diagnostics);
+        Ok(())
+    }
+
```

Comment:
I know this is a pattern that we have in other linter-groups too, but I don't think it's actually necessary to test a preview-rule. It's only necessary if the rule depends on any other preview-only behavior